src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1027,6 +1027,68 @@ impl Cage {
         }
     }
 
+    //------------------------------------FCHDIR SYSCALL------------------------------------
+    
+    pub fn fchdir_syscall(&self, fd: i32) -> i32 {
+        let mut path_string = String::new();
+        let unlocked_fd = self.filedescriptortable[fd as usize].read();
+        let mut first_iteration = true;
+
+        if let Some(filedesc_enum) = &*unlocked_fd {
+            match filedesc_enum {
+                File(normalfile_filedesc_obj) => {
+                    let mut inodenum = normalfile_filedesc_obj.inode;
+                    loop{
+                        let mut thisinode = match FS_METADATA.inodetable.get_mut(&inodenum) {
+                            Some(inode) => inode,
+                            None => {
+                                return syscall_error(Errno::ENOTDIR, "fchdir", "valid inodenum");
```

Comment:
- Since the inode itself is invalid should `EBADF` be returned or `ENOTDIR`?
- Can we make the error message "invalid inodenum" instead for more clarity?